### DISCUSSION TITLE
Is it possible to add an "accordion" functionality to PnP Search Results?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#2754] [Q&A] [answered]
Hi. Pardon for the seeming deluge of questions.
I am so enamored by this incredible work ( PnP Search) and its sheer power.... I digress.

I am hoping to replace FAQs which we currently use with PnP Search. Naturally.
The salient advantage the solution currently has is the "Accordion" feature.
It provides the users the ability to stay on the page and see the "description or content" without being directed to a display page for a given list item.
Is there way to implement this with PnP Search?
Thank you,
ted

**Comment by kasperbolarsen** [comment, marked as answer]
Perhaps you could use the master - details like functionality where you connect two Search Results web parts?

**Reply by tedtninety** [reply]
Thank you @kasperbolarsen . If I may ask, where may I find more information about "connecting two Search Results web parts", specifically for the purpose of integrating "Accordion" feature?  ( I realize now, it's collapsible content)
I scoured the web last night and I did manage to find information about "collapsible component".

<pnp-collapsible>[¶](https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/#pnp-collapsible)
Description: expand/collapse the collapsible-content content when the text with data-group-name is clicked.

I am trying to figure out how to integrate that into the "custom code" section.

**Reply by FranckyC** [reply]
Hi @tedtninety, If you simply need to display content as collapsible inside your template, you can use the buitlin [`<pnp-collapsible>` ](https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/#pnp-collapsible)component. More info here: https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/

**Reply by kasperbolarsen** [reply]
Connecting two search results web parts: https://franckcornu.com/post/modern-search-focus-item-selection/

As far as I know PnP modern search does not contain any option for an accordion layout so you might have to make it your self or find a woraround

**Reply by tedtninety** [reply]
@FranckyC  Oh Wow. Geeking out again here Thank you for your amazing work!!
That sounds great! I will research how to incorporate that. Cheers!

@kasperbolarsen  Thank you so very much. It will be wonderful to learn how to connect two search web parts.

**Reply by tedtninety** [reply]
Hi @FranckyC !! It works perfectly.  Thank you.
May I please ask, is there a way to have the header results without the extra white spaces between them. Having a number of returns creates a lengthy scroll.
Thank you.

**Reply by FranckyC** [reply]
Hi @tedtninety you could probably inject you own `<style>` in the template itself I guess. Didn't try but should work.

**Reply by tedtninety** [reply]
Thank you @FranckyC . :)  I have been trying to isolate the CSS for what adds that extra space but it has been a mystery thus far, and I'm usually quite adept with my CSS.

I opened a [thread](https://github.com/microsoft-search/pnp-modern-search/discussions/2761).
Thank you as always.

**Comment by PowerProvoker** [comment]
Howdy folks, funnily enough I'm after something very similar. Following what @FranckyC has suggested here - would there be a way to display a relationship between a Document Set and the contained Documents within? Where by each <pnp-collabsible> is a Document Set and then use a <ul> to display the Documents within that set as the <template id="collapsible-content"> item?
